tensorboard/data/server/data_compat.rs

```diff
@@ -38,6 +38,12 @@ pub(crate) mod plugin_names {
     pub const GRAPH_RUN_METADATA: &str = "graph_run_metadata";
     pub const GRAPH_RUN_METADATA_WITH_GRAPH: &str = "graph_run_metadata_graph";
     pub const GRAPH_KERAS_MODEL: &str = "graph_keras_model";
+    pub const HISTOGRAMS: &str = "histograms";
+    pub const TEXT: &str = "text";
+    pub const PR_CURVES: &str = "pr_curves";
+    pub const HPARAMS: &str = "hparams";
+    pub const MESH: &str = "mesh";
```

Comment:
Oh, interesting. I hadn’t planned to accommodate `mesh` because it
wasn’t obvious to me that the data fit well into the existing storage
classes. It’s not clearly `Tensor` because the data may be arbitrarily
large, and plausibly over our recommended limits. It’s not clearly
`BlobSequence` because it’s the wrong shape—though we could faithfully
reinterpret-cast without too much trouble, since all dimensions but one
are constant.

But I’m happy to just make it `Tensor` as you do here. I’m not sure how
widely used it is in conjunction with environments that would actually
care about the size limit (e.g., TensorBoard.dev uploads), so it may
just be a non-issue. As you note, the plugin backend still uses the
multiplexer, but we can fix that.

Reply:
Yeah—in the past, I think, we’d agreed that it was “reasonable” to
assume that text would be small, even though it’s not bounded. The
unbounded-size tensor is also approximately what I was thinking, as
“basically sugar for a length-1 blob sequence, but we handle the
serialization for you”. From that lens, I don’t know that we actually
need a separate data type—`emit_tensor(..., as_blob=True)` or something
might suffice.

Happy to leave as is for now.

Reply:
True, I guess it basically is just sugar for a length-1 blob sequence, so it could be persisted as such.  But I think we'd probably need a new data type at the `DataProvider` level if we wanted to both handle the Tensor serialization and also support a streaming read, right?